### PR TITLE
Containers refactoring

### PR DESCRIPTION
**Pull request by tsvisabo**
Export some of the containers logic to `utils`

* Bitset
* hash to int64 map
* hash to hash_set map - just find function for now

### CODE REVIEW

File: utils/hash_containers.c [hunk 1]
```diff
@@ -73,12 +73,81 @@ bool hash_set_contains(hash_set_t *set, flex_trit_t *hash) {
...
+  map_entry->value = value;
+  HASH_ADD(hh, *map, hash, FLEX_TRIT_SIZE_243, map_entry);
   return RC_OK;
 }
+
+void hash_int_map_free(hash_int_map_t *map) {
+  hash_to_int_value_map_entry *curr_entry = NULL;
+  hash_to_int_value_map_entry *tmp_entry = NULL;
+
+  // Cleanup CWRatings Map
```
Comment (thibault-martinez) [inline utils/hash_containers.c]: remove

File: utils/hash_containers.c [hunk 2]
```diff
@@ -73,12 +73,81 @@ bool hash_set_contains(hash_set_t *set, flex_trit_t *hash) {
...
+  HASH_ITER(hh, *map, curr_entry, tmp_entry) {
+    HASH_DEL(*map, curr_entry);
+    free(curr_entry);
+  }
+}
+
+/*
+ *  hash_to_indexed_hash_set_map_t
+ */
+bool hash_to_hash_set_map_contains(hash_to_indexed_hash_set_map_t *map,
```
Comment (thibault-martinez) [inline utils/hash_containers.c]: Can we name the type and the functions with the same prefix for consistency ?

File: utils/hash_containers.h [hunk 3]
```diff
@@ -31,21 +35,63 @@ typedef struct hash_queue_entry_s {
...
+
+/*
+ * Int Map operations
+ */
+
+typedef struct hash_to_int_value_map_entry {
+  flex_trit_t hash[FLEX_TRIT_SIZE_243];
+  int64_t value;
+  UT_hash_handle hh;
+} hash_to_int_value_map_entry;
```
Comment (thibault-martinez) [inline utils/hash_containers.h:66]: Can we call it `hash_int_map_entry_t` for consistency with the other types ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: utils/hash_containers.h (hunk 3 above)
Comment (tsvisabo) [reply utils/hash_containers.h:66]: Done

File: utils/hash_containers.c (hunk 2 above)
Comment (tsvisabo) [reply utils/hash_containers.c]: done

File: utils/hash_containers.c (hunk 1 above)
Comment (tsvisabo) [reply utils/hash_containers.c]: done

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**